engine/crates/engine-v2/src/engine/execute/single.rs

```diff
@@ -59,29 +62,56 @@ impl<R: Runtime> Engine<R> {
 
             // After recording all operation metadata
             tracing::debug!("Executed operation");
-            response
+
+            (response, on_operation_response_output)
         }
         .instrument(span.clone())
         .await
     }
 }
 
 impl<'ctx, R: Runtime> PreExecutionContext<'ctx, R> {
-    async fn execute_single(mut self, request: Request) -> (Option<OperationMetricsAttributes>, Response) {
-        let operation_plan = match self.prepare_operation(request).await {
+    async fn execute_single(
+        mut self,
+        request: Request,
+    ) -> (Option<OperationMetricsAttributes>, Option<Vec<u8>>, Response) {
+        let mut operation_info = hooks::Operation::builder();
+
+        let operation_plan = match self.prepare_operation(request, &mut operation_info).await {
             Ok(operation_plan) => operation_plan,
-            Err((metadata, response)) => return (metadata, response),
+            Err((metadata, response)) => return (metadata, None, response),
         };
 
-        let metrics_attributes = Some(operation_plan.metrics_attributes.clone());
-        let response = if matches!(operation_plan.ty(), OperationType::Subscription) {
-            RequestErrorResponse::bad_request_but_well_formed_graphql_over_http_request(
+        let metrics_attributes = operation_plan.metrics_attributes.clone();
+
+        if matches!(operation_plan.ty(), OperationType::Subscription) {
+            let response = RequestErrorResponse::bad_request_but_well_formed_graphql_over_http_request(
                 "Subscriptions are only suported on streaming transports. Try making a request with SSE or WebSockets",
-            )
+            );
+
+            (Some(metrics_attributes), None, response)
         } else {
-            self.execute_query_or_mutation(operation_plan).await
-        };
+            if let Some(ref name) = metrics_attributes.name {
+                operation_info.set_name(name.clone());
```

Comment:
This damn call here requires by-value transfer of the name. I tried to fight it a while, but decided just to pass this as a string to get it compiling.